File: futures/clients/websocket/node/run.example.js
require('dotenv').config();

const {
  FuturesPrivateWebSocketClient,
  PRIVATE_EVENTS
} = require('./client');

const authMethod = (process.env.AUTH_METHOD || 'api_key').toLowerCase();
if (!['api_key', 'jwt'].includes(authMethod)) {
  throw new Error('AUTH_METHOD must be "api_key" or "jwt"');
}
const commonOptions = {
  baseUrl: process.env.WEBSOCKET_BASE_URL,
  subaccountId: process.env.SUBACCOUNT_ID || undefined
};

const authOptions = authMethod === 'jwt'
  ? {
      jwt: process.env.JWT_TOKEN,
      clientType: process.env.CLIENT_TYPE || 'api'
    }
  : {
      apiKey: process.env.API_KEY,
      secretKey: process.env.SECRET_KEY
    };

const client = new FuturesPrivateWebSocketClient({
  ...commonOptions,
  ...authOptions
});

client.on('connect', (socketId) => {
  console.log('Socket.IO connected:', socketId);
});
client.on('auth.ok', (details) => {
  console.log('Private stream authenticated:', details);
});
client.on('auth.error', (error) => {
  console.error('Private stream authentication failed:', error);
});
client.on('credential_error', (error) => {
  console.error('Unable to build authentication credentials:', error.message);
});
client.on('connect_error', (error) => {
  console.error('Connection failed:', error.message);
});
client.on('disconnect', (reason) => {
  console.log('Disconnected:', reason);
});

for (const eventName of PRIVATE_EVENTS) {
  client.on(eventName, (payload) => {
    console.log(eventName, payload);
  });
}

process.on('SIGINT', () => {
  client.disconnect();
  process.exit(0);
});

client.connect();

File: futures/clients/websocket/python/run_example.py
import os

from dotenv import load_dotenv

from client import FuturesPrivateWebSocketClient, PRIVATE_EVENTS


load_dotenv()

auth_method = os.getenv('AUTH_METHOD', 'api_key').lower()
if auth_method not in {'api_key', 'jwt'}:
    raise ValueError('AUTH_METHOD must be "api_key" or "jwt"')
common_options = {
    'base_url': os.getenv(
        'WEBSOCKET_BASE_URL',
        'https://sp-futuresws.zebpay.com',
    ),
    'subaccount_id': os.getenv('SUBACCOUNT_ID') or None,
}
auth_options = (
    {
        'jwt': os.getenv('JWT_TOKEN'),
        'client_type': os.getenv('CLIENT_TYPE', 'api'),
    }
    if auth_method == 'jwt'
    else {
        'api_key': os.getenv('API_KEY'),
        'secret_key': os.getenv('SECRET_KEY'),
    }
)

client = FuturesPrivateWebSocketClient(
    **common_options,
    **auth_options,
)


@client.on('connect')
def on_connect():
    print('Socket.IO connected')


@client.on('auth.ok')
def on_auth_ok(details):
    print('Private stream authenticated:', details)


@client.on('auth.error')
def on_auth_error(error):
    print('Private stream authentication failed:', error)


@client.on('connect_error')
def on_connect_error(error):
    print('Connection failed:', error)


@client.on('disconnect')
def on_disconnect(*args):
    print('Disconnected:', *args)


def make_event_handler(event_name):
    def handler(payload):
        print(event_name, payload)
    return handler


for private_event in PRIVATE_EVENTS:
    client.on(private_event, make_event_handler(private_event))


try:
    client.connect()
    client.wait_for_auth()
    client.wait()
except KeyboardInterrupt:
    pass
finally:
    client.disconnect()
